src/domains/migration/hooks/use-migration-transactions.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useMemo } from "react";
+import { Contracts } from "@ardenthq/sdk-profiles";
+import { ethers } from "ethers";
+import { useConfiguration, useMigrations } from "@/app/contexts";
+import { useLatestTransactions } from "@/domains/dashboard/hooks";
+import { migrationNetwork, migrationWalletAddress } from "@/utils/polygon-migration";
+
+export const useMigrationTransactions = ({ profile }: { profile: Contracts.IProfile }) => {
+	const { profileIsSyncing, profileIsRestoring } = useConfiguration();
+	const { migrations, storeTransaction } = useMigrations();
+
+	const { isLoadingTransactions, latestTransactions } = useLatestTransactions({
```

Comment:
this will grab the latest 100 transactions for an address, irregardless of whether that's to the burn wallet? 🤔 I'd consider using the search API in core to already filter for migration wallet transactions for the given wallets, that should result in less requests and have them all combined